data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/configuration/EntryConfig.java

```diff
@@ -5,22 +5,36 @@
 
 package org.opensearch.dataprepper.plugins.processor.configuration;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.AssertTrue;
 import jakarta.validation.constraints.NotEmpty;
+import org.opensearch.dataprepper.plugins.processor.GeoIPDatabase;
+import org.opensearch.dataprepper.plugins.processor.GeoIPField;
 
+import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 public class EntryConfig {
-    static final String DEFAULT_TARGET = "geolocation";
+    static final String DEFAULT_TARGET = "geo";
     @JsonProperty("source")
     @NotEmpty
     private String source;
 
     @JsonProperty("target")
     private String target = DEFAULT_TARGET;
 
-    @JsonProperty("fields")
-    private List<String> fields;
+    @JsonProperty("include_fields")
```

Comment:
I tend to think we should move these up a level. Part of the reason to combine the processing in a single processor is to simplify the configurations. If we move it to the top, then users only have to configure once.

Reply:
Let's discuss it. I think if we are moving this there is no use of entries, and user can configure multiple instances of geoip processor as we are only downloading data once.